### PR TITLE
Test that the peers send at most one alert

### PR DESCRIPTION
**Pull request by ekasper**
Duplicate alerts have happened, see
70c22888c1648fe8652e77107f3c74bf2212de36

### CODE REVIEW

File: test/ssl_test.c [hunk 1]
```diff
@@ -100,6 +100,19 @@ static int check_alerts(HANDSHAKE_RESULT *result, SSL_TEST_CTX *test_ctx)
...
+    /*
+     * TODO: separate warning and fatal alerts if necessary. Current tests don't
+     * exercise multiple alerts but in theory, warning alerts could precede a
+     * fatal alert.
+     */
+    if (result->client_num_alerts_sent > 1) {
+        fprintf(stderr, "Client sent %d alerts.\n", result->client_num_alerts_sent);
+        return 0;
+    }
+    if(result->server_num_alerts_sent > 1) {
```
Comment (richsalz) [inline test/ssl_test.c]: space nit.

**Comment by richsalz** [comment]
+1 assuming you fix the "if(" nit.

**Comment by ekasper** [comment]
Interestingly this failed on Travis a couple of times with 'Client sent 2 alerts' but is passing for me locally. Will need to investigate a bit more before pushing.

**Comment by ekasper** [comment]
Ah - it's enable-ssl3 that fails! I'm on it.

**Comment by ekasper** [comment]
In SSLv3 we do send a warning alert, so I had to resolve the TODO. PTAL.

To support warning alerts, we could improve this to take a comma-separated list of expected alerts, but for now that doesn't seem so important.

File: test/README.ssltest.md [hunk 2]
```diff
@@ -69,7 +69,9 @@ handshake.
   - InternalError - some other error
 
 * ExpectedClientAlert, ExpectedServerAlert - expected alert. See
-  `ssl_test_ctx.c` for known values.
+  `ssl_test_ctx.c` for known values. Note: the expected alert is currently
+  matched against the _last_ received alert (i.e., a fatal alert or a
+  close_notify). Warning alert expectations are not yet supported.
```
Comment (richsalz) [inline test/README.ssltest.md]: "are currently ignored" ?

**Comment by richsalz** [comment]
+1  one minor suggestion on comment, just that very minor, your choice.

**Comment by ekasper** [comment]
Clarified the comment.

**Pull request closed without merging**